### DISCUSSION TITLE
[markdownlint-rule-helpers] use of MarkdownIt.Token instead of import("markdownlint").MarkdownItToken is wrong

### DISCUSSION THREAD

**Discussion by DavidAnson** [DefinitelyTyped/DefinitelyTyped#68926] [Issues with a @types package]
Relevant file: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/markdownlint-rule-helpers/index.d.ts

For example, the `line` and `lineNumber` properties are not present on `MarkdownIt.Token`, but always present on `MarkdownItToken`, leading to incorrect/misleading errors in correctly-implemented code like: `Property 'line' does not exist on type 'Token'.`.

[a comment by an automated account is not shown]

**Comment by DavidAnson** [comment]
An easy way to see this is open the `markdownlint` project in VS Code, checkout the `next` branch, and remove the workarounds here: https://github.com/DavidAnson/markdownlint/blob/next/test/rules/any-blockquote.js

```text
'blockquote.map' is possibly 'null'.
'blockquote.map' is possibly 'null'.
Property 'lineNumber' does not exist on type 'Token'.
Property 'line' does not exist on type 'Token'.
```

**Reply by JoshuaKGoldberg** [reply]
👍 It's not surprising #66413 didn't get the types quite right. And you're much more familiar with the `markdownlint` codebase than me. I'd be happy to ✅ a fix PR!
